mesonbuild/mcompile.py

```diff
@@ -353,6 +357,8 @@ def run(options: 'argparse.Namespace') -> int:
         raise MesonException(
             f'Backend `{backend}` is not yet supported by `compile`. Use generated project files directly instead.')
 
+    mlog.log(mlog.green('INFO:'), 'calculating backend command to run:', join_args(cmd))
+    sleep(2)
```

Comment:
Maybe I am not reading this right, but doesn't this this introduce a 2 seconds pause for each invocation of `meson compile`?

Reply:
This is intentional in order to guarantee that the user actually sees the informational messages before they scroll off the screen.

The overhead of invoking python and importing the entirety of the meson codebase in order to access utility functions, argument parsing, vsenv handling, and more, is already significant. If you want to get down to the metal and eke out as much speed as you can, you were already running ninja directly -- ninja gives instant results, mcompile never did.

Reply:
This is a bad idea. Particularly when using the vs backend there's not really a good alternative to mcompile due to the odd project names meson generates (e.g. "22e3565@@postgres@exe.vcxproj"). And even if those get fixed, people won't be able to rely on it in docs etc for a few years. What's the point here, except to annoy people that use a feature that you're not a fan of?

Reply:
To put this into perspective, on my current potato, to perform a no-op build.

running `meson compile` can take 31 seconds to complete on a cold cache, vs. 2 seconds to run ninja.

On a warm cache, running the same command twice in a row, for the second run, meson compile takes 1 second vs. ninja taking 0.005 seconds.

ninja utterly destroys meson compile in performance, because no-op builds are *actually* instant, they don't take an entire second to produce results. It's not especially unlikely that Meson gets evicted from the cache and starts taking longer, either.

What are your thoughts on immediately proceeding specifically during the vs backend due to the complexity of msbuild names decreasing the usefulness of knowing the command?

Reply:
> ninja utterly destroys meson compile in performance, because no-op builds are actually instant, they don't take an entire second to produce results. It's not especially unlikely that Meson gets evicted from the cache and starts taking longer, either.

Sure. I agree that using ninja in hack-compile-test loops is better. But that's not really related to mcompile adding a 2s sleep.

> What are your thoughts on immediately proceeding specifically during the vs backend due to the complexity of msbuild names decreasing the usefulness of knowing the command?

Not favorable. When writing getting-started docs needing to branch on the used backend increases complexity unnecessarily. And people unnecessarily waiting for 2s longer just makes meson look bad.

Reply:
Well, I guess we're screwed, better quickly rewrite it in rust so people don't have to wait an eyeblink.

Reply:
> running meson compile can take 31 seconds to complete on a cold cache, vs. 2 seconds to run ninja.

Really? On my laptop no-op rebuild of the giant GStreamer project with all its subprojects takes 200ms with ninja, 700ms with `meson compile`. That's a significant increase but barely noticable in practice. I'm sure we can improve that, for example IIRC mcompile reads introspection json instead of pickle meson internal data. However now it takes 2.7s which is in my opinion unacceptable just for printing INFO that I fail to understand why it is even needed in the first place. In the rare case you care about the command it ran, just scroll up...